### ISSUE TITLE
[pickers] Module not found: Can't resolve '@mui/base/utils' in …/node_modules/@mui/x-date-pickers-pro/DateRangeCalendar

### ISSUE THREAD

**Issue by segullshairbutt** [mui/mui-x#9650]
### Steps to reproduce 🕹

When I am trying to migrate from `"@mui/x-date-pickers": "^5.0.12"` to `^v6.9.2` (both simple and pro) by following the migration guide at:
https://mui.com/x/migration/migration-pickers-v5/

I faced the following error when I started my React App:
`Module not found: Can't resolve '@mui/base/utils' in …/node_modules/@mui/x-date-pickers-pro/DateRangeCalendar`

I visited the installation guide of Date Pickers to find out if I am missing anything at:
https://mui.com/material-ui/getting-started/installation/

Following are the Steps which I followed to update from `v5` to `v6`:
1. Removed `@mui/x-date-pickers`, `@mui/x-date-pickers-pro`, and `@emotion/styled` from `package.json`
2. Installed the latest version of the date picker by using `yarn add @mui/x-date-pickers` and `yarn add @emotion/styled`
3. Start react app by hitting the `npm start` command

Following are some other details about my project.
`"react": "^17.0.2"`
`"react-scripts": "4.0.3"`

### Current behavior 😯

Showing error message when starting application after following the migration guide from `v5` to `v6`.

### Expected behavior 🤔

It must give either typescript errors or notify the things for breaking changes, not the missing module information from inside of date picker modules.
Or there must be details about this error to occur during the upgrade. 

### Context 🔦

During updating the date pickers from `v5` to `v6` by following the migration guide I faced an issue when tried to start my react app. Either it should be included in the migration guide or installation guide or is needed to be focused on.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
 System:
    OS: macOS 13.2.1
  Binaries:
    Node: 20.2.0 - /opt/homebrew/bin/node
    Yarn: 1.22.11 - /opt/homebrew/bin/yarn
    npm: 9.6.6 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 114.0.5735.198
    Edge: Not Found
    Safari: 16.3
```
</details>

**Comment by LukasTy** [comment]
@segullshairbutt Could you please clarify the `@mui/material` package version that you are using?
Are you not getting a peer dependency mismatch warning when running the installation?
We've bumped the minimum supported version to `5.8.6`.

**Comment by segullshairbutt** [comment]
@LukasTy Sure, but I am using `"@mui/material": "^5.13.7"` in my project.

**Comment by LukasTy** [comment]
And what is the resolved version of the `@mui/base` package in your `node_modules`?

**Comment by segullshairbutt** [comment]
Well, it wasn't automatically resolved I needed to add it manually by hitting `yarn add @mui/base` and resolved version is `"@mui/base": "^5.0.0-beta.6"`

**Comment by LukasTy** [comment]
That's strange, it's a direct dependency of `@mui/material` and thus should have been resolved and downloaded. 🤔 
Have you tried the usual module issues debugging steps?
- remove `node_modules` and try running `yarn install`
- if the above didn't help: remove both `node_modules` and `yarn.lock` and try running `yarn install`

**Comment by segullshairbutt** [comment]
Yeah, I tried both of them eventually now again but still had the same error without any change in a word.

**Comment by LukasTy** [comment]
Could you please provide a working example with the mentioned problem reproducible?
It's hard to pinpoint or even investigate the issue when there are so many unknowns.
If possible, having a codesandbox example would be great.
If that's not possible, at least a repository with all the relevant package versions used would go a long way. 🙏

**Comment by peter1997546** [comment]
I have encountered similar issue recently.
The package installation works fine 1-2 weeks ago

**Comment by michelengelen** [comment]
`@mui/base` isn't a dependency of `@mui/x-date-pickers-pro` anymore; the package doesn't reference it at all in current versions. As you found yourself, the underlying cause was `react-scripts` 4, which had a stale module resolution; upgrading to `react-scripts` 5 (or a newer bundler in general) resolves this class of error.

Closing as completed, since the dependency this error was about no longer exists in the current package graph.
